lib/kafka/protocol/message.rb

```diff
@@ -57,11 +57,12 @@ def decompress
         data = codec.decompress(value)
         message_set_decoder = Decoder.from_string(data)
         message_set = MessageSet.decode(message_set_decoder)
+        base_offset = offset - message_set.size + 1
```

Comment:
`base_offset` should not be `offset - message_set.size + 1`
should be `offset - last(message_set).offset`
because relative offsets may not be sequential for compacted topics.